swri_roscpp/include/swri_roscpp/topic_service_client.h

```diff
@@ -29,88 +29,159 @@
 #ifndef SWRI_ROSCPP_TOPIC_SERVICE_CLIENT_H_
 #define SWRI_ROSCPP_TOPIC_SERVICE_CLIENT_H_
 
-#include <rclcpp/rclcpp.hpp>
+#include <chrono>
+#include <map>
+#include <mutex>
 
-#include <boost/thread/mutex.hpp>
-#include <boost/uuid/random_generator.hpp>
-#include <boost/uuid/uuid_io.hpp>
+#include <rclcpp/rclcpp.hpp>
 
-#include <map>
+using namespace std::chrono_literals;
 
 namespace swri
 {
 template<class MReq, class MRes>
 class TopicServiceClientRaw
 {
 private:
-  typedef
-  boost::mutex request_lock_;
+  std::mutex request_lock_;
   std::shared_ptr<rclcpp::Subscription<MRes> > response_sub_;
   std::shared_ptr<rclcpp::Publisher<MReq> > request_pub_;
   std::shared_ptr<MRes> response_;
 
-  rclcpp::Duration timeout_;
+  std::chrono::nanoseconds timeout_;
   std::string name_;
   std::string service_name_;
+  bool internal_spin_;
 
   int sequence_;
 
 public:
-  TopicServiceClientRaw() :
-    timeout_(std::chrono::duration<float>(4.0)),
+  TopicServiceClientRaw(bool internal_spin=true) :
```

Comment:
I think with addition of this in the constructor then requires a non-default constructor for the `TopicServiceClient` class to control this flag